### PR TITLE
Fix build using WebGPU with MAIN_MODULE

### LINKED ISSUE #14266: Undefined symbol errors in WebGPU builds since 2.0.19
I was running on version 2.0.18 before which passes my WebGPU builds without errors, but after upgrading to the newly released 2.0.21 the builds doesn't pass anymore. All versions from 2.0.19 and later has the following errors when built with -s USE_WEBGPU=1:

```
error: undefined symbol: wgpuDeviceCreateComputePipelineAsync (referenced by top-level compiled C/C++ code)
warning: Link with `-s LLD_REPORT_UNDEFINED` to get more information on undefined symbols
warning: To disable errors for undefined symbols use `-s ERROR_ON_UNDEFINED_SYMBOLS=0`
warning: _wgpuDeviceCreateComputePipelineAsync may need to be added to EXPORTED_FUNCTIONS if it arrives from a system library
error: undefined symbol: wgpuDeviceCreateRenderPipelineAsync (referenced by top-level compiled C/C++ code)
warning: _wgpuDeviceCreateRenderPipelineAsync may need to be added to EXPORTED_FUNCTIONS if it arrives from a system library
error: undefined symbol: wgpuDevicePopErrorScope (referenced by top-level compiled C/C++ code)
warning: _wgpuDevicePopErrorScope may need to be added to EXPORTED_FUNCTIONS if it arrives from a system library
error: undefined symbol: wgpuDevicePushErrorScope (referenced by top-level compiled C/C++ code)
warning: _wgpuDevicePushErrorScope may need to be added to EXPORTED_FUNCTIONS if it arrives from a system library
error: undefined symbol: wgpuDeviceSetDeviceLostCallback (referenced by top-level compiled C/C++ code)
warning: _wgpuDeviceSetDeviceLostCallback may need to be added to EXPORTED_FUNCTIONS if it arrives from a system libraryerror: undefined symbol: wgpuRenderBundleEncoderDrawIndexedIndirect (referenced by top-level compiled C/C++ code)
warning: _wgpuRenderBundleEncoderDrawIndexedIndirect may need to be added to EXPORTED_FUNCTIONS if it arrives from a system library
error: undefined symbol: wgpuRenderBundleEncoderDrawIndirect (referenced by top-level compiled C/C++ code)
warning: _wgpuRenderBundleEncoderDrawIndirect may need to be added to EXPORTED_FUNCTIONS if it arrives from a system library
error: undefined symbol: wgpuSwapChainConfigure (referenced by top-level compiled C/C++ code)
warning: _wgpuSwapChainConfigure may need to be added to EXPORTED_FUNCTIONS if it arrives from a system library

Error: Aborting compilation due to previous errors
```

The issue can be reproduced by the following hello world project:

main_module.cpp:
```
#include <cstdio>
int main()
{
        printf("hello world\n");
        return 0;
}
```

CMakeLists.txt:
```
cmake_minimum_required (VERSION 3.16)

project(app VERSION 0.1 LANGUAGES CXX)

set(CMAKE_CXX_STANDARD 17)
set(CMAKE_CXX_STANDARD_REQUIRED True)
set(CMAKE_BUILD_TYPE Debug)

add_compile_options(-Werror -Wfatal-errors -fPIC)

add_executable (main_module main_module.cpp)

set_target_properties(main_module PROPERTIES LINK_FLAGS "-O2 -s MAIN_MODULE=1 -s EXPORT_ALL=1 -s USE_WEBGPU=1 -s WASM=1 -s ASYNCIFY=1 -s ASSERTIONS=1 -s DISABLE_EXCEPTION_CATCHING=0 -s ALLOW_MEMORY_GROWTH=1")
```

Build with:
```
if [ ! -d "build" ]; then
    mkdir build
fi

pushd build > /dev/null
emcmake cmake -GNinja ..
ninja
popd > /dev/null
```

### PR DESCRIPTION
**Pull request by kainino0x**
Fixes #14266

### CODE REVIEW

File: src/library_webgpu.js [hunk 1]
```diff
@@ -854,6 +888,38 @@ var LibraryWebGPU = {
     return WebGPU.mgrQuerySet.create(device["createQuerySet"](desc));
   },
 
+  wgpuDeviceCreateRenderBundleEncoder: function(deviceId, descriptor) {
```
Comment (kainino0x) [inline src/library_webgpu.js:891]: Moved without changes

File: src/library_webgpu.js [hunk 2]
```diff
@@ -546,6 +546,40 @@ var LibraryWebGPU = {
...
+
+  wgpuDevicePopErrorScope: function(deviceId, callback, userdata) {
+    abort('unimplemented (TODO)');
+  },
+
+  wgpuDeviceSetDeviceLostCallback: function(deviceId, callback, userdata) {
+    abort('unimplemented (TODO)');
+  },
+
+#if MINIMAL_RUNTIME
```
Comment (kainino0x) [inline src/library_webgpu.js:561]: Moved without changes

**Review by kainino0x — COMMENTED** [review]
@austinEng PTAL

**Review by austinEng — APPROVED** [review]
LGTM - I didn't realize we had a few of these new functions unimplemented

**Comment by kainino0x** [comment]
@juj could I ask you for an approving review?

**Review by juj — APPROVED** [review]
lgtm

**Pull request merged**